File: day:39.2/findMajority.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        hash ={}
        [result, max] = [0, 0];
        for i in range(len(nums)):
            num = nums[i]
            if(num in hash):
                hash[num] += 1;
            else:
                hash[num] = 1;
            if(hash[num] > max):
                result = num;
                max = hash[num];
    
        return result;

File: day:39.2/findMajority.kt
class Solution {
    fun majorityElement(nums: IntArray): Int {
        var hash = mutableMapOf<Int, Int>();
        var result: Int = 0;
        var max = 0;
        
        for(num in nums) {
            if(num in hash) {
                hash.put(num, hash.get(num)!! + 1);
            }
            else {
                hash.put(num, 1);
            }
            if(hash.get(num)!! > max) {
                result = num;
                max = hash.get(num)!!;
            }
        }
        return result;        
    }
}
